addon/helpers/page-title.js

```diff
@@ -1,13 +1,30 @@
 import { scheduleOnce } from '@ember/runloop';
 import { inject as service } from '@ember/service';
 import Helper from '@ember/component/helper';
-import { set, get } from '@ember/object';
+import { get } from '@ember/object';
+import { reads } from '@ember/object/computed';
 import { guidFor } from '@ember/object/internals';
 import { assign } from '@ember/polyfills';
 import { getOwner } from '@ember/application';
 
-function updateTitle(tokens) {
-  set(this, 'title', tokens.toString());
+function updateTitle(document, tokens, isFastBoot) {
+  const toBeTitle = tokens.toString();
+
+  if (isFastBoot) {
+    let childNodes = document.head.childNodes;
+    for (let i = 0; i < childNodes.length; i++) {
+      let node = childNodes.item(i);
+      if (node.tagName.toLowerCase() === 'title') {
+        document.head.removeChild(node);
+      }
+    }
```

Comment:
Why do we need to remove the prior `<title>` tags that we find? In FastBoot the only thing that would be here that might include the prior `<title>` would have been someones `head.hbs` or something, the static `<title>` that is included in `app/index.html` would **not** be overridden by this.

Reply:
It is not about the index.html static `title` tag, its about nested routes page-title invocations. We either lookup existing `title` element or clean up them all, so nested route titles like `A | B | C` - would render correctly.

What we can do is lookup existing `title` element we add on our own and remove all textNodes from it but end result is kind of the same.

Reply:
Hmm, maybe we can track the current element and just update its contents on subsequent invocations?

Reply:
https://github.com/ember-fastboot/simple-dom/blob/master/packages/%40simple-dom/document/src/node.ts#L38 - I have not found any update methods on simple-dom `node` implementation to do this.

Reply:
I don't think we can track either, because if we imagine nested routes:

`A -> B -> C ... n`

Then each of them has `page-title` invocation in template, each of them triggers `run.schedule afterRender` queueing and the fact that the update method is separate method here, is only thing that helps runloop to run it just once, although N number of invocations tried to initiate it again before afterRender queue ran.

Not sure how we could keep track of the element if don't know which of the helper instances has latest element reference.

Reply:
maybe we need to put the update logic into a service and the helper only updates some datastructure that keeps the order and builds out a string

Reply:
This actually will not be a problem if we properly wait for transition to finish for all title updates.